assets/opstools/AppBuilder/classes/platform/views/ABViewFormConnect.js

```diff
@@ -348,13 +354,28 @@ module.exports = class ABViewFormConnect extends ABViewFormConnectCore {
                   // get the ui id of this component that matches the link Object
                   let fieldToCheck;
                   fieldsDefs.forEach((fdefs) => {
+                     // if the field has a custom foreign key we need to store it
+                     // so selectivity later can know what value to get, otherwise
+                     // we just get the uuid of the record
                      if (
+                        fdefs.settings.isCustomFK &&
+                        fdefs.settings.indexField != "" &&
                         fdefs.settings.linkObject &&
                         fdefs.settings.linkType == "one" &&
                         fdefs.settings.linkObject ==
                            formElementsDefs.settings.linkObject
                      ) {
-                        fieldToCheck = fdefs.id;
+                        let customFK = view.application.definitionForID(
+                           fdefs.settings.indexField
+                        );
+                        fieldToCheck = fdefs.id + ":" + customFK.columnName;
```

Comment:
we should check to make sure `customFK` was actually returned from `.definitionForID()`